### PR TITLE
Cleanup tickers

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

File: clock/clock.go [hunk 1]
```diff
@@ -15,11 +16,13 @@ func AlignedTickLossy(period time.Duration) <-chan time.Time {
...
-			unix := time.Now().UnixNano()
-			diff := time.Duration(period - (time.Duration(unix) % period))
+			now := time.Now()
+			nowUnix := now.UnixNano()
+			diff := period - (time.Duration(nowUnix) % period)
+			ideal := now.Add(diff)
 			time.Sleep(diff)
 			select {
-			case c <- time.Now():
+			case c <- ideal:
```
Comment (replay) [inline clock/clock.go]: What's the purpose of this change? Is it to make the time pushed into the channel more closely aligned with the exact desired time.
This means that the time which gets pushed into the channel might be slightly different from the time when it gets pushed into the channel. 
I'm not sure which is the preferred behavior, the old one or the new one. Is there a specific reason why this is better?

Comment (Dieterbe) [reply clock/clock.go]: > What's the purpose of this change? Is it to make the time pushed into the channel more closely aligned with the exact desired time.

yes

> This means that the time which gets pushed into the channel might be slightly different from the time when it gets pushed into the channel.

yes. though even with the previous code, by the time the value makes it over the channel, and the receiver reads it out, the clock will have advanced. it's a law of nature that can't be changed.

> I'm not sure which is the preferred behavior, the old one or the new one. Is there a specific reason why this is better?

the new one seems preferably to me. because:
1) with the old code, on a loaded system, due to goroutine pre-emption the sleep may take, let's say a second longer than intended. that would mean timestamps are no longer round. e.g. instead of 00:00:00 UTC you may get 00:00:01 UTC
2) it makes it consistent with AlignedTickLossless
